test/widgets/message_list_test.dart

```diff
@@ -981,4 +981,70 @@ void main() {
       });
     });
   });
+
+  group('TypingStatus', () {
```

Comment:
separate nit: where possible we try to keep tests in the same order as the code they're testing, to help with navigating between tests and code under test

Here MarkAsReadWidget is already out of order, so moving that would be a good small prep commit. Then this can go next to that.